lib/CircuitBreaker.js

```diff
@@ -1,10 +1,150 @@
+ * @property {Object[]} global - global probe configurations (circuit breaks everything)
+ * @property {Object.<string, Object[]>} topic - Per topic probe configurations
+ * @property {Object.<string, Object[]>} location - Per location probe configurations
+ */
+
+/**
+ * returns a list of relevant topics to the bucket processor
+ * @param {Object} lcConfig lifecycle configuration
+ * @param {Object} repConfig replication configuration
+ * @param {Object} locations  location configuration
+ * @returns {string[]} list of topics
+ */
+function getAllTopics(lcConfig, repConfig, locations) {
+    const topics = [
+        lcConfig.objectTasksTopic,
+        repConfig.dataMoverTopic,
+    ];
+    Object.keys(locations).forEach(locationName => {
+        if (locations[locationName].isCold) {
+            topics.push(lcConfig.coldStorageArchiveTopicPrefix + locationName);
+        }
+    });
+    return topics;
+}
+
+/**
+ * Adapts the templated circuit breaker configuration to the bucket processor
+ * Invalid probes are ignored
+ * @param {Object} cbConf templated circuit breaker configuration
+ * @param {Object} lcConfig lifecycle configuration
+ * @param {Object} repConfig replication configuration
+ * @param {Object} locations location configuration
+ * @returns {BucketProcessorCircuitBreakerConfig} circuit breaker configuration
+ */
+function extractBucketProcessorCircuitBreakerConfigs(cbConf, lcConfig, repConfig, locations) {
+    const configs = {
+        global: [],
+        topic: {},
+        location: {},
+    };
+
+    if (!cbConf || !cbConf.probes) {
+        return configs;
+    }
+
+    cbConf.probes.forEach(probe => {
+        if (probe.type !== 'prometheusQuery') {
+            configs.global.push(probe);
+            return;
+        }
+        // with clause is optional and can be used to specify the filtering on
+        // the topic and location, along with specifying if a probe should be
+        // used for transition and expiration.
+        // Query syntax:
+        // "with(topic=topicName,location=locationName,transition=true,expiration=true) and
+        //      PROM_METRIC_NAME{topic='${topic}',location='${location}'}"
+        // Note: should either specify the topic or the location, not both
+        // Behaviour:
+        // - no with clause + templated topic: creates a circuit breaker for each topic
+        // - with clause with a specific topic + templated topic: creates a circuit breaker for the specified topic
+        // - with clause with a specific location + templated location: creates a circuit breaker for the location
+        const withClause = probe.query.match(/with\((.*?)\)\sand\s/);
+
+        const withTopicTemplate = probe.query.includes('${topic}');
+        const withLocationTemplate = probe.query.includes('${location}');
+
+        const topic = withClause && withClause[1].match(/topic=(.*?)(,|$)/);
+        const location = withClause && withClause[1].match(/location=(.*?)(,|$)/);
```

Comment:
ideally we would want to properly parse/validate the "content" of the withClause : i.e.
* split by comma to get the individual "filters"
* check each filter key is valid : topic, location, transition, expiration
* check the operator (`=` for now)

though it may do for now, kind of in a pinch... can you create a ticket nonetheless to address this?